XCUITests/DownloadFilesTests.swift

```diff
@@ -0,0 +1,172 @@
+        waitUntilPageLoad()
+        // Verify that the context menu prior to download a file is correct
+        app.webViews.staticTexts[testFileName].tap()
+        app.webViews.buttons["Download"].tap()
+        waitforExistence(app.tables["Context Menu"])
+        XCTAssertTrue(app.tables["Context Menu"].staticTexts[testFileName].exists)
+        XCTAssertTrue(app.tables["Context Menu"].cells["download"].exists)
+        app.buttons["Cancel"].tap()
+    }
+
+    func testDownloadOptionFromBrowserTabMenu() {
+        navigator.goto(BrowserTabMenu)
+        navigator.goto(HomePanel_Downloads)
+        XCTAssertTrue(app.tables["DownloadsTable"].exists)
+    }
+
+    func testDownloadFile() {
+        downloadFile(fileName: testFileName, numberOfDownlowds: 1)
+        navigator.goto(BrowserTabMenu)
+        navigator.goto(HomePanel_Downloads)
+
+        waitforExistence(app.tables["DownloadsTable"])
+        // There should be one item downloaded. It's name and size should be shown
+        checkTheNumberOfDownloadedItems(items: 1)
+        XCTAssertTrue(app.tables.cells.staticTexts[testFileName].exists)
+        XCTAssertTrue(app.tables.cells.staticTexts[testFileSize].exists)
+    }
+
+    func testDeleteDownloadedFile() {
+        downloadFile(fileName: testFileName, numberOfDownlowds: 1)
+        navigator.goto(BrowserTabMenu)
+        navigator.goto(HomePanel_Downloads)
+        waitforExistence(app.tables["DownloadsTable"])
+
+        deleteItem(itemName: testFileName)
+        waitforNoExistence(app.tables.cells.staticTexts[testFileName])
+
+        // After removing the number of items should be 0
+        checkTheNumberOfDownloadedItems(items: 0)
+    }
+
+    func testShareDownloadedFile() {
+        downloadFile(fileName: testFileName, numberOfDownlowds: 1)
+        navigator.goto(BrowserTabMenu)
+        navigator.goto(HomePanel_Downloads)
+        app.tables.cells.staticTexts[testFileName].swipeLeft()
+
+        XCTAssertTrue(app.tables.cells.buttons["Share"].exists)
+        XCTAssertTrue(app.tables.cells.buttons["Delete"].exists)
+
+        app.tables.cells.buttons["Share"].tap()
+        waitforExistence(app.otherElements["ActivityListView"])
+        if iPad() {
+            app.otherElements["PopoverDismissRegion"].tap()
+        } else {
+            app.buttons["Cancel"].tap()
+        }
+    }
+
+    func testLongPressOnDownloadedFile() {
+        downloadFile(fileName: testFileName, numberOfDownlowds: 1)
+        navigator.goto(BrowserTabMenu)
+        navigator.goto(HomePanel_Downloads)
+
+        waitforExistence(app.tables["DownloadsTable"])
+        app.tables.cells.staticTexts[testFileName].press(forDuration: 2)
+        waitforExistence(app.otherElements["ActivityListView"])
+        if iPad() {
+            app.otherElements["PopoverDismissRegion"].tap()
+        } else {
+            app.buttons["Cancel"].tap()
+        }
+     }
+
+    private func downloadFile(fileName: String, numberOfDownlowds: Int) {
```

Comment:
@npark-mozilla Please let me know your view on this, if this is now more similar to what you commented. Thanks!